addOns/jwt/src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanner.java

```diff
@@ -0,0 +1,252 @@
+        newValue = JWTUtils.extractingJWTFromParamValue(newValue);
+
+        if (!JWTUtils.isTokenValid(newValue)) {
+            LOGGER.info("Token: " + newValue + " is not a valid JWT token.");
+            return;
+        }
+        // Sending request to save actual response and then compare it with new response
+        try {
+            sendAndReceive(msg);
+        } catch (IOException e) {
+            LOGGER.error(e);
+            return;
+        }
+
+        JWTTokenBean jwtTokenBean;
+        try {
+            jwtTokenBean = JWTTokenBean.parseJWTToken(newValue);
+        } catch (JWTException e) {
+            LOGGER.error("Unable to parse JWT Token", e);
+            return;
+        }
+
+        if (!JWTConfiguration.getInstance().isIgnoreClientConfigurationScan()) {
+            if (performAttackClientSideConfigurations(msg, param, jwtTokenBean, value)) {
+                return;
+            }
+            this.decreaseRequestCount();
+        }
+        performAttackServerSideConfigurations(msg, param, jwtTokenBean, value);
+    }
+
+    public boolean isStop() {
+        return super.isStop() || (this.maxRequestCount.get() <= 0);
+    }
+
+    public void decreaseRequestCount() {
+        this.maxRequestCount.decrementAndGet();
+    }
+
+    /**
+     * performs attack to find if client side configuration for JWT token is proper.
+     *
+     * @param msg
+     * @param param
+     * @param jwtTokenBean
+     * @return {@code true} if the vulnerability is found, {@code false} otherwise.
+     */
+    private boolean performAttackClientSideConfigurations(
+            HttpMessage msg, String param, JWTTokenBean jwtTokenBean, String value) {
+        return new ClientSideAttack(this, param, msg).execute();
+    }
+
+    /**
+     * performs attack to find JWT implementation weaknesses like weak key usage or other types of
+     * attacks.
+     *
+     * @param msg
+     * @param param
+     * @param jwtTokenBean
+     * @return {@code true} if the vulnerability is found, {@code false} otherwise.
+     */
+    private boolean performAttackServerSideConfigurations(
+            HttpMessage msg, String param, JWTTokenBean jwtTokenBean, String value) {
+
+        // TODO there are scenarios where base64 encoded secrete is used in JWT. more
+        // information in
+        // below link
+        // https://stackoverflow.com/questions/58044813/how-to-create-a-jwt-in-java-with-the-secret-base64-encoded
+        boolean result = new ServerSideAttack(jwtTokenBean, this, param, msg, value).execute();
+        if (!result) {
+            result = new BruteforceAttack(jwtTokenBean, this, param, msg).execute();
+        }
+        return result;
+    }
+
+    /**
+     * Waits some time to check if token is expired and then execute the attack. TODO need to
+     * implement it.
+     *
+     * @param jwtTokenBean
+     * @param msg
+     * @param param
+     * @return
+     */
+    private boolean checkExpiredTokenAttack(
```

Comment:
Not sure how we can implement this (ie whether JWT Validator is honouring the expiry time or not) as Active scanner has a fixed time limit to run and JWT token can have expiry of 30 minutes too so it is not easy to implement. **Need to think more.**
@thc202 @kingthorin @psiinon  please suggest if you guys have idea regarding this  problem.